lib/phoenix/sync/write.ex

```diff
@@ -0,0 +1,970 @@
+defmodule Phoenix.Sync.Write do
+  @moduledoc """
+  Provides [optimistic write](https://electric-sql.com/docs/guides/writes)
+  support for Phoenix- or Plug-based apps.
+
+  The client uses some local database and syncs changes into this database from
+  the server using [Electric](https://electric-sql.com/).
+
+  Local writes are performed on the local database using
+  [TanStack/optimistic](https://github.com/TanStack/optimistic) which POSTs
+  transactions to the Elixir/Phoenix application.
+
+  A transaction is a list of "mutations", a series of `INSERT`, `UPDATE` and
```

Comment:
Generally, I think we need to expand on the rationale for this module better. The point of this integration is to ingest these transactions in a way that leverages your existing auth and context functions. I think we need to say this more clearly and we need to set out an example up front that shows this.

I.e.: say you have an existing Auth plug and an existing changes function, this is how you wire this in to re-use to auth and validate changes coming in through this endpoint. I think showing this re-use of existing code to ingest transactions is the key thing to communicate around the rationale for this code.

Reply:
There's enough of a difference between controller-based auth and this mutation-based version that it will always require some kind of mapping/refactoring to make the validations usable from both contexts. Not sure how much of that is up to us to explain and how much is just about the dev understanding the domain. 

But yes, I've focused on the API docs, not the rationale